Specify number of deployments to retain when pruning during release (#1230)
hashicorp/waypoint: website/content/docs/lifecycle/release.mdx

In this scenario, it is expected that either the deployment platform makes
the deployments available to traffic or this is handled externally.

## Deployment Pruning

During the release stage, Waypoint will prune unreleased deployments.

> By default, Waypoint keeps the past two deployments at all times for

Reviewer: ```suggestion
By default, Waypoint keeps just one previous deployment for
```
